src/gui/accountsettings.cpp

```diff
@@ -688,14 +688,22 @@ void AccountSettings::slotAccountStateChanged()
     /* Allow to expand the item if the account is connected. */
     ui->_folderList->setItemsExpandable(state == AccountState::Connected);
 
-    /* check if there are expanded root items, if so, close them, if the state is different from being Connected. */
     if (state != AccountState::Connected) {
+        /* check if there are expanded root items, if so, close them */
         int i;
         for (i = 0; i < _model->rowCount(); ++i) {
             if (ui->_folderList->isExpanded(_model->index(i)))
                 ui->_folderList->setExpanded(_model->index(i), false);
         }
+    } else if (_model->isDirty()) {
+        // If we connect and have pending changes, show the list.
+        doExpand();
```

Comment:
however, does that show the "changes" if they are deep in the list?  doExpand only expand one item.  Maybe it stays expanded as before because the model was not reset?

(But even if it does not, I think it's still better than not showing it.)